src/batch/src/task/context.rs

```diff
@@ -67,6 +68,52 @@ pub trait BatchTaskContext: Clone + Send + Sync + 'static {
     fn mem_usage(&self) -> usize;
 
     fn create_executor_mem_context(&self, executor_id: &str) -> Option<MemoryContextRef>;
+    fn get_stop_flag(&self) -> Arc<StopFlag>;
+
+    fn get_stop_flag_ref(&self) -> &StopFlag;
+}
+
+/// We already have `shut_down` sender to stop execution.
+/// `StopFlag` used in executor to let it stop execute more timely when it can't receive message
+/// timely.
+#[derive(Clone)]
+pub struct StopFlag {
+    stop_flag: Arc<AtomicBool>,
+    stop_reason: Arc<RwLock<Option<String>>>,
```

Comment:
This could cause concurrent bugs. For example: 

1. thread A sets flag = true
2. thread B gets flag = true
3. thread B gets reason = None  <-- unexpected! 
4. thread A sets reason = "something"

Is it possible to substitute an atomic reference for the `AtomicBool`?

Reply:
I used store(Release) to gurantee that write before store(Relase) will be see if the store(Release) is visable.🤔
https://github.com/risingwavelabs/risingwave/blob/b43d0bd119d958417d5b2306694f96ed92873012/src/batch/src/task/context.rs#L99

[ref](https://marabos.nl/atomics/memory-ordering.html#:~:text=A%20happens%2Dbefore%20relationship%20is%20formed%20when%20an%20acquire%2Dload%20operation%20observes%20the%20result%20of%20a%20release%2Dstore%20operation.%20In%20this%20case%2C%20the%20store%20and%20everything%20before%20it%2C%20happened%20before%20the%20load%20and%20everything%20after%20it.)

Reply:
I see. LGTM